jupyterhub_moss/spawner.py

```diff
@@ -72,13 +72,9 @@ class MOSlurmSpawner(SlurmSpawner):
     ).tag(config=True)
 
     @traitlets.validate("partitions")
-    def _validate_partitions(self, proposal):
-        # Set add_to_path if missing in jupyter_environments
-        partitions = deepcopy(proposal["value"])
-        for partition in partitions.values():
-            for env in partition["jupyter_environments"].values():
-                env.setdefault("add_to_path", True)
-                env.setdefault("prologue", "")
+    def _validate_partitions(self, proposal: dict) -> dict[str, dict]:
+        partitions = proposal["value"]
+        PartitionsTrait.parse_obj(partitions)  # Validation
```

Comment:
Now only use `parse_obj` for validation but do not convert result back to `dict` since it would lack `prologue`.

Reply:
If you need to exclude `prologue` only at a particular place, I think it makes more sense to keep your previous version where you exclude where it is needed rather than setting the `exclude` globally.

Sorry for misleading you :smile:

Reply:
There is no need to update the `partitions` dict when validating it, it's anyway converted again when it needs to be retrieved.
The validation here is only to fail early if there is an issue with it.